### PR TITLE
Fixed bug on outcome version increment [SUBS-438]

### PR DESCRIPTION
**Pull request by fpenim**
Now using an `Integer` as the validation outcome version with increments of 1.

### CODE REVIEW

File: validator-aggregator/src/main/java/uk/ac/ebi/subs/validator/aggregator/OutcomeDocumentService.java [hunk 1]
```diff
@@ -35,11 +35,11 @@ public boolean isLatestVersion(String submissionId, String entityUuid, double th
...
 
         if (validationOutcomes.size() > 0) {
-            List<Double> doubleVersions = validationOutcomes.stream()
-                    .map(validationOutcome -> Double.valueOf(validationOutcome.getVersion()))
+            List<Integer> versions = validationOutcomes.stream()
+                    .map(validationOutcome -> Integer.valueOf(validationOutcome.getVersion()))
                     .collect(Collectors.toList());
 
-            double max = Collections.max(doubleVersions);
+            int max = Collections.max(versions);
```
Comment (rolando-ebi) [inline validator-aggregator/src/main/java/uk/ac/ebi/subs/validator/aggregator/OutcomeDocumentService.java]: Collections.max() should work on collections of Strings so I'm not sure if it's necessary to convert the version string to a number first i.e Collections.max(["1.111", "0.999445", "1.2"]) should return "1.2".

Comment (fpenim) [inline validator-aggregator/src/main/java/uk/ac/ebi/subs/validator/aggregator/OutcomeDocumentService.java]: Ok, I'll try to do that.

File: validator-coordinator/src/main/java/uk/ac/ebi/subs/validator/coordinator/OutcomeDocumentService.java [hunk 2]
```diff
@@ -43,18 +43,18 @@ public String getVersion(String submissionId, String entityUuid) {
...
-            List<Double> doubleVersions = validationOutcomes.stream()
-                    .map(validationOutcome -> Double.valueOf(validationOutcome.getVersion()))
+            List<Integer> versions = validationOutcomes.stream()
+                    .map(validationOutcome -> Integer.valueOf(validationOutcome.getVersion()))
                     .collect(Collectors.toList());
 
-            double max = Collections.max(doubleVersions);
-            String version = String.valueOf(max + 0.1);
+            int max = Collections.max(versions);
+            String version = String.valueOf(max + 1);
```
Comment (ke4) [inline validator-coordinator/src/main/java/uk/ac/ebi/subs/validator/coordinator/OutcomeDocumentService.java]: Please, change also the method documentation to increment it by 1 and not 0.1.

File: validator-aggregator/src/main/java/uk/ac/ebi/subs/validator/aggregator/OutcomeDocumentService.java [hunk 3]
```diff
@@ -31,16 +31,16 @@ public boolean updateValidationOutcome(EntityValidationOutcome validationOutcome
...
-            List<Double> doubleVersions = validationOutcomes.stream()
-                    .map(validationOutcome -> Double.valueOf(validationOutcome.getVersion()))
+            List<Integer> versions = validationOutcomes.stream()
+                    .map(validationOutcome -> validationOutcome.getVersion())
                     .collect(Collectors.toList());
 
-            double max = Collections.max(doubleVersions);
-            if (max > thisOutcomeVersion) {
+            int max = Integer.valueOf(Collections.max(versions));
+            if (max > Integer.valueOf(thisOutcomeVersion)) {
```
Comment (ke4) [inline validator-aggregator/src/main/java/uk/ac/ebi/subs/validator/aggregator/OutcomeDocumentService.java]: We don't need the Integer.valueOf() method calls

File: validator-status-flipper/src/main/java/uk/ac/ebi/subs/validator/flipper/OutcomeDocumentService.java [hunk 4]
```diff
@@ -41,16 +39,16 @@ public boolean updateValidationOutcome(String uuid) {
...
-            List<Double> doubleVersions = validationOutcomes.stream()
-                    .map(validationOutcome -> Double.valueOf(validationOutcome.getVersion()))
+            List<Integer> versions = validationOutcomes.stream()
+                    .map(validationOutcome -> validationOutcome.getVersion())
                     .collect(Collectors.toList());
 
-            double max = Collections.max(doubleVersions);
-            if (max > thisOutcomeVersion) {
+            int max = Integer.valueOf(Collections.max(versions));
+            if (max > Integer.valueOf(thisOutcomeVersion)) {
```
Comment (ke4) [inline validator-status-flipper/src/main/java/uk/ac/ebi/subs/validator/flipper/OutcomeDocumentService.java]: We don't need the Integer.valueOf() method calls

**Pull request merged**